dasharo-security/tpm-support.robot

```diff
@@ -138,3 +131,40 @@ TPM003.003 Check TPM Physical Presence Interface (Windows)
 #    Skip If    not ${tpm_support}    TPM003.004 not supported
 #    Skip If    not ${tests_in_ubuntu_support}    TPM003.004 not supported
 # TODO: https://docs.dasharo.com/unified-test-documentation/dasharo-security/200-tpm-support/#tpm003004-change-active-pcr-banks-with-tpm-ppi-firmware
+
+
+*** Keywords ***
+Validate Any TPM
+    [Documentation]    Checks for TPM major version, and validates it.
+    ${tpm_ver}=    Execute Command In Terminal    cat /sys/class/tpm/tpm0/tpm_version_major
+    IF    '${tpm_ver}' == '2'
+        Detect Or Install Package    tpm2-tools
+        ${out}=    Execute Command In Terminal    tpm2_pcrread
+        Should Contain    ${out}    sha1:
+        Should Contain    ${out}    sha256:
+    ELSE IF    '${tpm_ver}' == '1'
+        Detect Or Install Package    tpm-tools
+        ${out}=    Execute Command In Terminal    tpm_selftest
```

Comment:
If we validate TPm by checking PCRs, we should also do it the same way for TPM1.2. TPm1.2 PCRs are also populated in sysfs `/sys/class/tpm/tpm0/device/pcrs`

Reply:
TPM 2.0 also reports PCRs in sysfs for quite some time, in different format, but the same way works for TPM 1.2 (except it has only `sha1` obviously):

```bash
$ ls /sys/class/tpm/tpm0/pcr-sha*
/sys/class/tpm/tpm0/pcr-sha1:
0  10  12  14  16  18  2   21  23  4  6  8
1  11  13  15  17  19  20  22  3   5  7  9

/sys/class/tpm/tpm0/pcr-sha256:
0  10  12  14  16  18  2   21  23  4  6  8
1  11  13  15  17  19  20  22  3   5  7  9
$ cat /sys/class/tpm/tpm0/pcr-sha256/4
E61988A7E3DD2C2F69C3226E4904E69D8A43BA0A35E3BC0930A8A9E0E7CAAD21
```

Reply:
Yup, valid point. No need to install tpm2-tools just for this